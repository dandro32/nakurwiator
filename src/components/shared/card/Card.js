import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';


const Card = ({card}) => {
  return (
    <div className="card">
      <img className="card-img-top" src={card.imgSrc} />
      <div className="card-body">
        <h5 className="card-title">{card.title}</h5>
        <p className="card-text">{card.desc}</p>
        <Link to={card.href} className="btn btn-primary">
          {card.hrefTitle}
        </Link>
      </div>
    </div>
  );
};

Card.propTypes = {
  card: PropTypes.object.isRequired
};

export default Card;