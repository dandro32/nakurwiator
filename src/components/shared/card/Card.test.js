import React from 'react';
import Card from './Card';
import {HashRouter as Router} from 'react-router-dom';
import TestRenderer from 'react-test-renderer';

describe('Card component tests', ()=> {
  let result;
  const mockCard = {
    title: 'title',
    imgSrc: 'imgSrc',
    desc: 'desc',
    href: 'href',
    hrefTitle: 'hrefTitle'
  };
  beforeEach(() => {
    result = TestRenderer.create(
      <Card card={mockCard} />
    ).toTree();

  });

  it('renders correctly', () => {
    console.log(result);
    expect(result).toMatchSnapshot();
  });
});