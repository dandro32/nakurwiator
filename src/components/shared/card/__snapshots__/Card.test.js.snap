// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component tests renders correctly 1`] = `undefined`;
